Update doc for desktop integration and rendering backend (#1703)
f3d-app/f3d: doc/user/DESKTOP_INTEGRATION.md

F3D can be integrated in the desktop experience.

F3D now supports rendering in sandboxed environments. To enable thumbnail generation in sandboxed environments (e.g., Nautilus), add the `--rendering-backend` option to specify EGL as the backend:

For Example:

> Exec=f3d --rendering-backend=egl --output=%o --resolution=%s,%s %i

Reviewer: thats nice, but this is specific to linux, so you may want to put that in the Linux section

> 

Reviewer: the limitations you need to remove is in (LIMITATIONS_AND_TROUBLESHOOTING.md) part.

Reviewer: please remove this added line

Reviewer: this line has still not been remove

sudo update-desktop-database ~/.local/share/applications
```

> F3D supports rendering in environments with limited graphical environnements, including sandboxes or without Xorg. In most cases, the default behavior will automatically detect which capabilities are available and use the right rendering backend, it is possible to specify it using `--rendering-backend` CLI option.

Reviewer 2: Avoid repetition and specify "running" for Xorg
```suggestion
F3D supports rendering in environments with limited graphical contexts, including sandboxes or without Xorg running. In most cases, the default behavior will automatically detect which capabilities are available and use the right rendering backend, it is possible to specify it using `--rendering-backend` CLI option.
```

Reviewer: you can take @Meakk suggestion into account, either by modifying as suggested or clicking on "commit suggestion".

If `libEGL.so` is available, you can use: `--rendering-backend=egl`.
If `libOSMesa.so` is available, you can use: `--rendering-backend=osmesa`.

> You can use that feature for thumbnail generation in sandboxed environments (e.g., Nautilus), by modifying the `f3d.thumbnailer` file like this:

Reviewer 2: @mwestphal is it the right approach? Should we add the option in the thumbnail global config file?

Reviewer: currently being discussed here: https://discord.com/channels/1046005690809978911/1306190530270724128

```
Exec=f3d --rendering-backend=egl --output=%o --resolution=%s,%s %i

> ```

Reviewer: So lets not focus on the `f3d.thumbnailer` file but instead focus on adding a specific line in the config file. You could write something like:

You can use that feature for thumbnail generation whenver needed (e.g., Nautilus), by modifying the [thumbnail config file](CONFIGURATION_FILE.md), eg: `~/.config/f3d/thumbnail.json` :

```
    ".*": {
         "rendering-backend": "egl"
   },
```

Author: Can i know where i have to add this line i.e., in desktop integration md or somewhere else

> Please note this change will be overriden if you reinstall F3D.

Reviewer 2: ```suggestion
Please note this change will be overridden if you reinstall F3D.
```

> Please note this change will be overridden if you reinstall F3D.

Reviewer: remove this line

sudo update-desktop-database ~/.local/share/applications
```

F3D supports rendering in environments with limited graphical contexts, including sandboxes or without Xorg running. In most cases, the default behavior will automatically detect which capabilities are available and use the right rendering backend, it is possible to specify it using `--rendering-backend` CLI option.

If `libEGL.so` is available, you can use: `--rendering-backend=egl`.
If `libOSMesa.so` is available, you can use: `--rendering-backend=osmesa`.

> You can use that feature for thumbnail generation whenver needed (e.g., Nautilus), by modifying the thumbnail config file, eg:`~/.config/f3d/thumbnail.json `:

Reviewer: add a link around config file:

```
[config file](CONFIGURATION_FILE.md)
```

```
    ".*": {
         "rendering-backend": "egl"

>    },

Reviewer: small typos:

```
    ".*": 
    {
         "rendering-backend": "egl"
    }
 ```

sudo update-desktop-database ~/.local/share/applications
```

F3D supports rendering in environments with limited graphical contexts, including sandboxes or without Xorg running. In most cases, the default behavior will automatically detect which capabilities are available and use the right rendering backend, it is possible to specify it using `--rendering-backend` CLI option.

If `libEGL.so` is available, you can use: `--rendering-backend=egl`.
If `libOSMesa.so` is available, you can use: `--rendering-backend=osmesa`.

> You can use that feature for thumbnail generation whenver needed (e.g., Nautilus), by modifying the [thumbnail config file](CONFIGURATION_FILE.md), eg:`~/.config/f3d/thumbnail.json `:

Reviewer: whenever

> ```

Reviewer: add `javascript` after the three quotes for syntax highlighting

> ```

Reviewer: we are using `javascript` in the rest of the doc, I think json is not supported.

sudo update-desktop-database ~/.local/share/applications
```

F3D supports rendering in environments with limited graphical environnements, including sandboxes or without Xorg. In most cases, the default behavior will automatically detect which capabilities are available and use the right rendering backend, it is possible to specify it using `--rendering-backend` CLI option.

If `libEGL.so` is available, you can use: `--rendering-backend=egl`.

> If `libOSMesa.so` is available, you can user: `--rendering-backend=osmesa`.

Reviewer: user -> use

You can use that feature for thumbnail generation in sandboxed environments (e.g., Nautilus), by modifying the `f3d.thumbnailer` file like this:

> ```Exec=f3d --rendering-backend=egl --output=%o --resolution=%s,%s %i ```

Reviewer: ```suggestion
```
Exec=f3d --rendering-backend=egl --output=%o --resolution=%s,%s %i 
```
```

Author: Are you referring like this way:

Exec= f3d
      --rendering-backend=eg1
      --output=%o --resolution=%s,%s %i

If you have any issues, refer to the [troubleshooting](LIMITATIONS_AND_TROUBLESHOOTING.md) part.

> F3D now supports rendering in sandboxed environments. To enable thumbnail generation in sandboxed environments (e.g., Nautilus), add the `--rendering-backend` option to specify EGL as the backend:

Reviewer: F3D supports rendering in environments with limited graphical environnements, including sandboxes or without Xorg. In most cases, the default behavior will automatically detect which capabilities are available and use the right rendering backend, it is possible to specify it using `--rendering-backend` CLI option.

If `libEGL.so` is available, you can use: `--rendering-backend=egl`.
If `libOSMesa.so` is available, you can user: `--rendering-backend=osmesa`.

You can use that feature for thumbnail generation in sandboxed environments (e.g., Nautilus), by modifying the `f3d.thumbnailer` file like this:

```
Exec=f3d --rendering-backend=egl --output=%o --resolution=%s,%s %i
```

Please note this change will be overriden if you reinstall F3D.

If you have any issues, refer to the [troubleshooting](LIMITATIONS_AND_TROUBLESHOOTING.md) part.

> F3D supports rendering in environments with limited graphical environnements, including sandboxes or without Xorg. In most cases, the default behavior will automatically detect which capabilities are available and use the right rendering backend, it is possible to specify it using --rendering-backend CLI option.

Reviewer: use "`" around `rendering-backend`

> If libEGL.so is available, you can use: --rendering-backend=egl.

Reviewer: use \`

> If libOSMesa.so is available, you can user: --rendering-backend=osmesa.

Reviewer: use `

You can use that feature for thumbnail generation in sandboxed environments (e.g., Nautilus), by modifying the f3d.thumbnailer file like this:

> Exec=f3d --rendering-backend=egl --output=%o --resolution=%s,%s %i

Reviewer: use

\'\'\'

before and after the Exec line

Please note this change will be overriden if you reinstall F3D.

> Some file managers (eg: Nautilus) are using sandboxing for thumbnails, which the F3D standard binary release does not support as it needs access to the Xorg server to render anything.

Reviewer: remove this

There is no support for thumbnails on MacOS, the .dmg binary release provides automatic file openings.

## Limitations

> - TIFF files are excluded from the thumbnailer to avoid conflicts with other better 2D thumbnailers.

Reviewer: this limitation is still true and should not be removed.

## Limitations

- TIFF files are excluded from the thumbnailer to avoid conflicts with other better 2D thumbnailers.

> Contributor

Reviewer: remove this
